### DISCUSSION TITLE
Require Statement Error Handling

### DISCUSSION THREAD

**Discussion by a-znk** [wevm/wagmi#1728] [Q&A]
If i'd like to display the custom string from a `require()` statement on my Dapp, what is the best way to extract that error using a Wagmi hook? (if I can)

For example, here is the contract mint function:

```
function mint(uint256 _mintAmount) public payable mintCompliance(_mintAmount) mintPriceCompliance(_mintAmount) {
    require(!paused, 'The contract is paused!');

    _safeMint(_msgSender(), _mintAmount);
  }
```

I'd like to display `The contract is paused!` on the screen for a user on the front-end.

**Comment by challet** [comment]
I guess that would be through the [ContractMethodRevertedError](https://github.com/wagmi-dev/wagmi/blob/wagmi%400.12.1/packages/core/src/errors.ts#L196..L233) error where there is an `errorMessage` property. Try to inspect the error you get when the transaction fails because of this unmet condition.

**Reply by a-znk** [reply]
Thank you for the reply, though I'm afraid I'm a bit out of my depth here. Are you able to expand any further please?

**Reply by challet** [reply]
I'd like to but that won't be easy without to see the code from where the`mint`method is called. Have you tried anything yet ?

**Reply by a-znk** [reply]
This is how I'm currently calling `mint`

```
  const {
    config,
    error: prepareError,
    isError: isPrepareError,
  } = usePrepareContractWrite({
    ...contractConfig,
    functionName: "mint",
    args: [mintAmount],
    overrides: {
      value: costData?.mul(mintAmount),
    },
    onError(error) {
      console.log(error);
    },
  });

  const {
    error: mintError,
    isError: isMintError,
    data: mintData,
    write: mint,
    isLoading: isMintLoading,
    isSuccess: isMintStarted,
  } = useContractWrite(config);
```

**Reply by challet** [reply]
Does the logged error contain the message you're looking for ?

**Reply by a-znk** [reply]
Not quite... I want to display the error from the smart contract require statement. This displays the error from Ethers directly I think... I can get a more readable error if i do this

```
            {(isPrepareError || isMintError) && (
              <div className={mintStyles.mint_error}>
                Error: {(prepareError || mintError)?.reason}
              </div>
            )}
```

But then I get this TS error

> Property 'reason' does not exist on type 'Error'.ts(2339)

Is there some sort of interface I could add to fix the TS error, or is this bad practice in general?

**Reply by challet** [reply]
I don't think there is a reason property. What does it show when you log it into the console to inspect it ?
Note that some `console.log` implementation uses the `toString` method of the object if it exists and for an error that will display only its `message` property. You may need to use [`console.dir()`](https://developer.mozilla.org/en/docs/Web/API/console/dir) instead.

**Reply by a-znk** [reply]
`reason` displaying this message: 

> `ERROR: INSUFFICIENT FUNDS FOR INTRINSIC TRANSACTION COST`

`message` displays a _much_ longer message which includes a lot of information about the error, the gas fee, a URL to Ethers, and more.

I'm pretty lost here. I've seen @tmm mention `reason` in separate posts before.

**Reply by challet** [reply]
It looks like there is this other error before to reach the actual revert error you're looking for. Isn't there anything related to a revert in the lot of information you're mentioning ?

**Reply by challet** [reply]
> I'm pretty lost here. I've seen @tmm mention `reason` in separate posts before.

Errors from ethers indeed have a `reason` property. But using wagmi I think they are parsed and thrown as a new one, like [here](https://github.com/wagmi-dev/wagmi/blob/ef380d9c6d51ae0495b9c35925d2843c75d97fd4/packages/core/src/actions/contracts/multicall.ts#L137). Actually, it seems that for this specific type (`ContractMethodRevertedError`) it is happening [only from a multicall](https://github.com/wagmi-dev/wagmi/search?q=ContractMethodRevertedError) … so maybe it is a messy as with ethers :/
